src/components/GanttChart.tsx

```
@@ -375,32 +375,37 @@
+							// For now, let's assume existingReactTask is found.
+						}
+						return {
+							...(existingReactTask || {}), // Spread existing task (has custom props), provide empty object fallback
+							id: ganttTask.id,     // Ensure these are from Gantt's current state post-drag
+							text: ganttTask.text,
+							// Ensure start_date and end_date are Date objects before formatting
+							start_date: formatDate(new Date(ganttTask.start_date as any)),
+							end_date: formatDate(new Date(ganttTask.end_date as any)),
+							duration: ganttTask.duration,
+							parent: ganttTask.parent, // parent ID from Gantt
+							progress: ganttTask.progress,
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)